aries/aries_vcx_core/src/wallet2/mod.rs

```diff
@@ -0,0 +1,393 @@
+use async_trait::async_trait;
+use derive_builder::Builder;
+#[cfg(feature = "vdrtools_wallet")]
+use indy_api_types::domain::wallet::Record as IndyRecord;
+use serde::{Deserialize, Serialize};
+
+use self::entry_tag::EntryTags;
+use crate::{errors::error::VcxCoreResult, wallet::structs_io::UnpackMessageOutput};
+
+#[cfg(feature = "vdrtools_wallet")]
+pub mod indy_wallet;
+
+pub mod entry_tag;
+pub mod utils;
+
+pub struct Key {
+    pub pubkey_bs58: String,
+}
+
+#[derive(Debug, Default, Clone, Builder)]
+pub struct Record {
+    pub category: String,
+    pub name: String,
+    pub value: String,
+    #[builder(default = "EntryTags::default()")]
+    pub tags: EntryTags,
+}
+
+#[cfg(feature = "vdrtools_wallet")]
+impl From<IndyRecord> for Record {
+    fn from(ir: IndyRecord) -> Self {
+        Self {
+            name: ir.id,
+            category: ir.type_,
+            value: ir.value,
+            tags: ir.tags.into(),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct DidData {
+    did: String,
+    verkey: String,
+}
+
+pub enum SearchFilter {
+    JsonFilter(String),
```

Comment:
I agree that using json is unfortunate, but these changes are about introducing a new backward-compatible trait that would allow us a transition to different wallet implementation(s). These changes are not about improving the existing wallet implementation. 

Even though I could come up with a new type to use here, it would still need to be turned into json at some point because that is what vdrtools [wallet requires](https://github.com/hyperledger/aries-vcx/blob/main/aries/misc/legacy/libvdrtools/src/controllers/non_secrets.rs#L357) and that would just result in more overhead.